### DISCUSSION TITLE
Can I use the HTML embed element?

### DISCUSSION THREAD

**Discussion by bpgs** [datenstrom/community#524] [Ask a question] [answered]
I have been using yellow for a long time. In my blog I publish charts that are created with https://www.goat1000.com/svggraph.php. I use the HTML tag <embed.../> to insert SVG charts. Until version yellow 0.8.6 everything was fine. But after the update to version 0.8.11 or now 0.8.12 <embed.../> creates only an empty p-tag and no chart is visible.

**Comment by schulle4u** [comment, marked as answer]
Yellow uses a HTML whitelist to filter out potentially malicious code. The `embed` tag currently is not in this list, however I'm sure there are alternative elements for this case. Can you please provide some example code?

**Comment by bpgs** [comment]
In my blog I use code snippets similar to the description of the framework:
https://www.goat1000.com/svggraph-using.php
option 2 and option 4 are ok, option 1 und 3 don't work if I use them with yellow
The exact code on my test page ist

```
## Einfügen mit embed

<embed src="/charts-3.6/demo-01.php" type="image/svg+xml" width="600" height="400"
  pluginspage="http://www.adobe.com/svg/viewer/install/" />

## Einfügen mit iframe

<iframe src="/charts-3.6/demo-01.php" type="image/svg+xml" width="600" height="400"></iframe>

## Einfügen mit object

<object data="/charts-3.6/demo-01.php" width="600" height="400" type="image/svg+xml" />

## Einfügen mit img

<img src="/charts-3.6/demo-01.php" width="600" height="400" type="image/svg_xml" />


```
When I use this code with **yellow 0.8.6** or in plain HTML everything is ok.

**Comment by bpgs** [comment]
I have inserted
`"embed"`
in line 2387 of my core.php. Now everything is fine. is there any reason for omitting this tag (and <object)?

**Comment by markseuffert** [comment]
As Steffen said, the HTML whitelist protects websites from malicious code. Unfortunately, we cannot add `<embed> <object>` to the whitelist, because they can potentially be abused for cross side scripting (XSS). See also https://owasp.org/www-community/xss-filter-evasion-cheatsheet.

**Comment by schulle4u** [comment]
Moreover there's simply no longer a need for these elements as they mostly have been used for web technologies such as Java applets and Flash that fortunately are unsupported or banned in most modern browsers, not to mention mobile devices.
